Orange/widgets/widget.py

```diff
@@ -508,6 +508,14 @@ def closeContext(self):
         reinitializing the user interface (e.g. combo boxes) with the new
         data.
         """
+        # Clear schema-only settings when *closing* the context
+        # FIXME: Sadly, schema-only settings aren't cleared when non-contextual
```

Comment:
But what if the _same instance_ of widget but passed different data, as described [here](https://github.com/biolab/orange3/pull/2301/files#diff-2281ecd412ed7edd749aca4486d1f338R779)?